delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/GlobalEventLog.scala

```diff
@@ -151,55 +150,54 @@ object GlobalEventLog {
       xas: Transactors
   ): GlobalEventLog[Id, S, Command, E, Rejection] = new GlobalEventLog[Id, S, Command, E, Rejection] {
 
-    override def stateOr[R <: Rejection](id: Id, notFound: => R): IO[R, S] = stateStore.get(id).toUIO.flatMap {
-      IO.fromOption(_, notFound)
+    override def stateOr[R <: Rejection](id: Id, notFound: => R): IO[S] = stateStore.get(id).flatMap {
+      IO.fromOption(_)(notFound)
     }
 
-    override def stateOr[R <: Rejection](id: Id, rev: Int, notFound: => R, invalidRevision: (Int, Int) => R): IO[R, S] =
-      stateMachine.computeState(eventStore.history(id, rev)).flatMap {
+    override def stateOr[R <: Rejection](id: Id, rev: Int, notFound: => R, invalidRevision: (Int, Int) => R): IO[S] =
+      stateMachine.computeState(eventStore.history(id, rev).translate(ioToTaskK)).toCatsIO.flatMap {
         case Some(s) if s.rev == rev => IO.pure(s)
         case Some(s)                 => IO.raiseError(invalidRevision(rev, s.rev))
         case None                    => IO.raiseError(notFound)
       }
 
-    override def evaluate(id: Id, command: Command): IO[Rejection, (E, S)] =
-      stateStore.get(id).toUIO.flatMap { current =>
+    override def evaluate(id: Id, command: Command): IO[(E, S)] =
```

Comment:
Actually that second statement probably won't work without [this](https://github.com/oleg-py/better-monadic-for) 😓 but in Scala 3 it is supported out of the box

Reply:
I definitely like the way the function is split up in your suggestion. I'd like to experiment with `EitherT` but I won't do it in this PR, as there is a lot of code I would like to refactor and I'll never finish if I start :)

Reply:
Note that in `EphemeralLog` that has already been migrated, there is no such thing so it would be better if we could have some consistency between the different logs (`EphemeralLog` is way simpler though)